### DISCUSSION TITLE
Why does the cursor size differ from one application to another?

### DISCUSSION THREAD

**Discussion by azizsayfiddin** [hyprwm/Hyprland#6840] [Q&A]
I set the the cursor theme using the nwg-look and the `hyprctl setcursor ...`.
[Demo](https://files.catbox.moe/mllbut.mp4)

**Comment by vaxerski** [comment]
https://wiki.hyprland.org/Hypr-Ecosystem/hyprcursor/

**Reply by azizsayfiddin** [reply]
How can I find out whether the hyprcursor is running or not?

**Reply by azizsayfiddin** [reply]
Because the cursor in the KeePassXC is still large.

**Reply by azizsayfiddin** [reply]
[Demo](https://files.catbox.moe/27tf24.webm)

**Reply by Nicolas-Saavedra** [reply]
Can you show a demo where the cursor is one size on one app, and another in another one, while running only a theme on Hyprcursor and not on XCursor? Maybe that can help to understand the issue. Just giving out ideas.

**Reply by azizsayfiddin** [reply]
> Can you show a demo where the cursor is one size on one app, and another in another one, while running only a theme on Hyprcursor and not on XCursor? Maybe that can help to understand the issue. Just giving out ideas.

[Here](https://files.catbox.moe/wz6ycy.webm)

**Reply by ghost** [reply]
Have you checked to see if this is a qt/gtk inconsistency? Maybe you just need to make sure you have the same cursor size configured across qt/gtk versions?

**Reply by azizsayfiddin** [reply]
I changed the gtk cursor size using the nwg-look. How can I change the qt cursor size?

**Reply by azizsayfiddin** [reply]
> Have you checked to see if this is a qt/gtk inconsistency? Maybe you just need to make sure you have the same cursor size configured across qt/gtk versions?

I set the catppuccin-sapphire [cursor](https://files.catbox.moe/y5rbbl.webm) in nwg-look, lxappearance, hyprland.conf.

**Reply by ghost** [reply]
lxappearance and nwg-look are both GTK, you probably don't need both. Using two different gtk theming apps could create issues if any of your settings weren't exactly the same between the two apps. What is your QT_QPA_PLATFORMTHEME env variable set to?

**Reply by azizsayfiddin** [reply]
> lxappearance and nwg-look are both GTK, you probably don't need both. Using two different gtk theming apps could create issues if any of your settings weren't exactly the same between the two apps. What is your QT_QPA_PLATFORMTHEME env variable set to?

It's set to qt5ct.

**Reply by ghost** [reply]
Do you have the same size set for XCURSOR and do you have qt5-wayland/qt6-wayland installed?

There is no direct setting for QT cursor size, it's usually picked up through the desktop environment or X resources.

**Reply by azizsayfiddin** [reply]
> Do you have the same size set for XCURSOR and do you have qt5-wayland/qt6-wayland installed?
> 
> There is no direct setting for QT cursor size, it's usually picked up through the desktop environment or X resources.

It becomes large when I set the XCURSOR_SIZE variable to 36, but it stays normal when set to 24. I can't understand why this happens. Is there any way to make it look similar to the cursor in the GTK apps?

**Reply by ghost** [reply]
The integer you used may be your problem - the sizes available vary by cursor theme. Standard sizes are 24, 32, 48, 64. 

Try those first and if you're still having trouble there are a few other things you can try.

**Reply by azizsayfiddin** [reply]
> The integer you used may be your problem - the sizes available vary by cursor theme. Standard sizes are 24, 32, 48, 64.
> 
> Try those first and if you're still having trouble there are a few other things you can try.

It works. How Do you know that? I changed size of the cursor from 36 to 32 in all configs.

**Reply by ghost** [reply]
I don't know anything. I just break things and try to fix them. 

I only know about this cursor size bs because I spent an unhealthy amount of time being pissed about it myself until I figured it out. 

And I read a lot. 

Happy I could help!

**Comment by azizsayfiddin** [comment]
How can I change the cursor size in the java applications e.g. minecraft?

**Reply by azizsayfiddin** [reply]
I tried to set the environment variables and java arguments, but it didn't help.
![image](https://github.com/user-attachments/assets/a54a0377-2d02-4d7b-a52e-83767d7aafbb)
![image](https://github.com/user-attachments/assets/27e405a7-2496-4340-8300-f1e063a2a9a0)

**Reply by ghost** [reply]
16 isn't a common size in cursor packages. Try 24, 32, 48, or 64.

**Comment by nktnet1** [comment]
Potentially related to #6983.

I solved a similar problem and have documented it here: https://github.com/hyprwm/Hyprland/issues/6320#issuecomment-2243109637

**Comment by ghost** [comment]
> Potentially related to #6983.
> 
> I solved a similar problem and have documented it here: https://github.com/hyprwm/Hyprland/issues/6320#issuecomment-2243109637
> 

Your issue was the same as this one, but that's because you both used cursor sizes that probably don't exist in your cursor package. 24, 32, 48, 64 are standard.

**Reply by nktnet1** [reply]
I don't think so - tried with both 24 and 32. It was more so a regression issue when migrating from wlroots-hyprland to aquamarine - this happened to many users (myself included) who did not use cursor themes prior to the update (see #6983).

**Reply by ghost** [reply]
Gotcha, didn't mean to downplay your issue - cursor size problems were just a major frustration for me and it ended up simply being an invalid size integer. Some newer cursor packages are scalable, but many aren't. Maybe this info will spare someone else an unnecessary headache.

**Comment by azizsayfiddin** [comment]
I enabled the glfw, now the cursor size is normal, but it's blurry. Do you have an idea why? I guess it's java rendering problem. [Recording](https://files.catbox.moe/lp05fh.mkv)
